gemrb/plugins/GUIScript/GUIScript.cpp

```diff
@@ -7873,7 +7882,12 @@ static PyObject* GemRB_UpdateVolume(PyObject* /*self*/, PyObject* args)
 {
 	int type = 3;
 	PARSE_ARGS(args, "i", &type);
-	core->GetAudioDrv()->UpdateVolume(type);
+	if (type == 1 || type == 3) {
+		core->GetMusicLoop().UpdateVolume();
+	}
+	if (type == 2 || type == 3) {
+		core->GetAmbientManager().UpdateVolume();
+	}
```

Comment:
You could use bit comparisons here to simplify. But this made me see a bug: the parameter is not optional any more, so we either need to fix the doc or use `"|i"`